lib/puppet/provider/katello_ssl_tool.rb

```diff
@@ -158,11 +158,11 @@ def self.target_path(file_name = '')
     end
 
     def build_path(file_name = '')
-      self.class.build_path(file_name)
+      self.class.build_path(file_name, resource[:build_dir])
     end
 
-    def self.build_path(file_name = '')
-      File.join("/root/ssl-build", file_name)
+    def self.build_path(file_name = '', build_dir)
```

Comment:
Yes, there is an over-use of `self` throughout that I would tackle as a re-factor after I do some other clean up first.

Reply:
I also did some of that when I tried to understand this PR. I think https://github.com/theforeman/puppet-certs/pull/418 is at least one clean up that can already happen.

Reply:
For example, I can now rip out all of the RPM generation and handling code since we've been running based on just the files from a deployment stand-point for a few release now. And that may end up cleaning out a lot of code. So I'd revisit re-factor after I do that.